src/application/components/ApplicationProcedureInfo.tsx

```diff
@@ -1,80 +1,110 @@
-        />
-        <Dialog.Content>
-          <ol className="ApplicationProcedureInfo__popup-steps">
-            <li>
-              {t(
-                'application.procedure.popup.stages.submission',
-                'The applicant submits their application.',
-              )}
-            </li>
-            <li>
-              {t(
-                'application.procedure.popup.stages.forwarding',
-                'The application is forwarded to the correct city department for processing.',
-              )}
-            </li>
-            <li>
-              {t(
-                'application.procedure.popup.stages.review',
-                'The city confirms if the area is eligible to be leased for the applied-for use case and requests further information from the applicant if necessary. Other parties relevant to the decision might be consulted at this stage as well.',
-              )}
-            </li>
-            <li>
-              {t(
-                'application.procedure.popup.stages.offer',
-                'If the city decides to lease the area to the applicant, a draft contract will be sent to them for commenting.',
-              )}
-            </li>
-            <li>
-              {t(
-                'application.procedure.popup.stages.decision',
-                'The city will make a formal lease decision and the lease contract will be signed.',
-              )}
-            </li>
-          </ol>
-        </Dialog.Content>
-      </Dialog>
-    </div>
+      </li>
+      <li>
+        {t(
+          'application.procedure.popup.stages.offer',
+          'If the city decides to lease the area to the applicant, a draft contract will be sent to them for commenting.',
+        )}
+      </li>
+      <li>
+        {t(
+          'application.procedure.popup.stages.decision',
+          'The city will make a formal lease decision and the lease contract will be signed.',
+        )}
+      </li>
+    </ol>
+  );
+
+  return (
+    <>
+      {!showOnlyContent ? (
+        <div className="ApplicationProcedureInfo">
+          <div id="dialog-target-ref" ref={dialogTargetRef} />
+          <Button
+            onClick={openModal}
+            variant="supplementary"
+            iconLeft={<IconQuestionCircle />}
+            ref={openInfoDialogButtonRef}
+          >
+            {t(
+              'application.procedure.link',
+              'Information about the application review process',
+            )}
+          </Button>
+          <Dialog
+            id="ApplicationProcedureInfoPopup"
+            className="ApplicationProcedureInfo__popup"
+            aria-labelledby="ApplicationProcedureInfoPopup_Header"
+            isOpen={isOpen}
+            close={closeModal}
+            closeButtonLabelText={t(
+              'application.procedure.popup.closeButton',
+              'Close',
+            )}
+            focusAfterCloseRef={openInfoDialogButtonRef}
+            targetElement={dialogTargetRef.current}
```

Comment:
Not sure if this is actually the correct line, but Typescript in the PR build gives this
```Error: src/application/components/ApplicationProcedureInfo.tsx(88,13): error TS2322: Type 'null' is not assignable to type 'HTMLElement | undefined'.```